api/src/main/java/org/apache/iceberg/expressions/CountNonNull.java

```diff
@@ -39,6 +39,9 @@ protected Long countFor(StructLike row) {
 
   @Override
   protected boolean hasValue(DataFile file) {
+    if (file.valueCounts() == null) {
```

Comment:
This logic seems duplicate, can we make a safeContainsKey in the base class (just like safeGet)